pubsub/gcloud/aio/pubsub/subscription.py

```diff
@@ -1,66 +1,106 @@
 import asyncio
-import logging
+import concurrent.futures
+import signal
+from asyncio import Future  # pylint: disable=ungrouped-imports
+from typing import Any
+from typing import Callable
+from typing import Optional
 
-import google.auth.exceptions
-import google.cloud.pubsub as pubsub
-import google.gax.errors
-import grpc
+from google.api_core import exceptions
+from google.cloud import pubsub
+from google.cloud.pubsub_v1.subscriber.message import Message
+from google.cloud.pubsub_v1.types import FlowControl
+from google.cloud.pubsub_v1.types import Scheduler
 
+from .utils import convert_google_future_to_concurrent_future
 
-log = logging.getLogger(__name__)
 
+class Subscription:
 
-class Subscription(pubsub.subscription.Subscription):
-    async def acknowledge(self, ack_ids, client=None, retries=0):
-        try:
-            return super().acknowledge(ack_ids, client=client)
-        except (google.auth.exceptions.TransportError,
-                google.gax.errors.RetryError):
-            # common intermittent communication errors
-            if not retries:
-                raise
-
-            await asyncio.sleep(1)
-
-        return await self.acknowledge(ack_ids, client=client,
-                                      retries=retries-1)
+    def __init__(self,
+                 subscription: str,
+                 *,
+                 loop: Optional[asyncio.AbstractEventLoop] = None
+                 ) -> None:
+        self.subscription = subscription
+        self._subscriber = pubsub.SubscriberClient()
 
-    def create_if_missing(self, client=None):
-        if self.exists(client=client):
-            return
+        if not loop:
+            loop = asyncio.get_event_loop()
+        self.loop = loop
 
+    def create_subscription(self,
+                            topic: str,
+                            *,
+                            ack_deadline_seconds: int = 5
```

Comment:
Tempted to just accept a `**kwargs` here and pass 'em directly to the Google client, so we don't have to keep this in sync with anything. Thoughts on that?

If so, let's link to [their docstring](https://github.com/googleapis/google-cloud-python/blob/11c72ade8b282ae1917fba19e7f4e0fe7176d12b/pubsub/google/cloud/pubsub_v1/gapic/subscriber_client.py#L236) rather than re-write our own.

Reply:
yeah that sounds good to me